rbx_xml/src/types/font.rs

```diff
@@ -0,0 +1,199 @@
+    Ok(())
+}
+
+fn read_content_inner(reader: &mut XmlEventReader<impl Read>) -> Result<String, DecodeError> {
+    match reader.expect_next()? {
+        XmlReadEvent::StartElement {
+            name,
+            attributes,
+            namespace,
+        } => match name.local_name.as_str() {
+            "null" => {
+                reader.expect_end_with_name("null")?;
+                Ok(String::new())
+            }
+            "url" => {
+                let value = reader.read_characters()?;
+                reader.expect_end_with_name("url")?;
+                Ok(value)
+            }
+            _ => {
+                let event = XmlReadEvent::StartElement {
+                    name,
+                    attributes,
+                    namespace,
+                };
+                Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event)))
+            }
+        },
+        event => Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event))),
+    }
+}
+
+fn read_content(reader: &mut XmlEventReader<impl Read>, tag: &str) -> Result<String, DecodeError> {
+    match reader.expect_next()? {
+        XmlReadEvent::StartElement {
+            name,
+            attributes,
+            namespace,
+        } => {
+            if name.local_name.as_str() != tag {
+                let event = XmlReadEvent::StartElement {
+                    name,
+                    attributes,
+                    namespace,
+                };
+                return Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event)));
+            }
+        }
+        event => return Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event))),
+    };
+
+    let value = read_content_inner(reader)?;
+
+    reader.expect_end_with_name(tag)?;
+
+    Ok(value)
+}
+
+impl XmlType for Font {
+    const XML_TAG_NAME: &'static str = "Font";
+
+    fn write_xml<W: Write>(&self, writer: &mut XmlEventWriter<W>) -> Result<(), EncodeError> {
+        write_content(writer, &self.family, "Family")?;
+
+        let weight: u16 = self.weight.clone().into();
+        let weight: i32 = weight.into();
+        writer.write_value_in_tag(&weight, "Weight")?;
+
+        let style = match self.style {
+            FontStyle::Normal => "Normal",
+            FontStyle::Italic => "Italic",
+            FontStyle::Other(_) => "Normal",
+        };
+        writer.write_tag_characters("Style", style)?;
+
+        if let Some(ref cached_face_id) = self.cached_face_id {
+            write_content(writer, cached_face_id, "CachedFaceId")?;
+        }
+
+        Ok(())
+    }
+
+    fn read_xml<R: Read>(reader: &mut XmlEventReader<R>) -> Result<Self, DecodeError> {
+        let family = read_content(reader, "Family")?;
+
+        let weight: i32 = reader.read_value_in_tag("Weight")?;
+        let weight: u16 = weight
+            .clamp(u16::MIN as i32, u16::MAX as i32)
+            .try_into()
+            .unwrap();
+        let weight = FontWeight::from(weight);
```

Comment:
Provided the suggestion for `FontWeight::from_u16` is adopted:
```suggestion
        let weight = FontWeight::from_u16(weight);
```